aws-distro-opentelemetry-node-autoinstrumentation/src/patches/aws/services/bedrock.ts

```diff
@@ -211,11 +211,156 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
       spanAttributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_MODEL] = modelId;
     }
 
+    if (request.commandInput?.body) {
+      const requestBody = JSON.parse(request.commandInput.body);
+      if (modelId.includes('amazon.titan')) {
+        if (requestBody.textGenerationConfig) {
+          const config = requestBody.textGenerationConfig;
+          spanAttributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_TEMPERATURE] = config.temperature;
+          spanAttributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_TOP_P] = config.topP;
+          spanAttributes[AwsSpanProcessingUtil.GEN_AI_REQUEST_MAX_TOKENS] = config.maxTokenCount;
```

Comment:
does each of these input attribute have a default value if user doesn't set it. Eg, could `config.maxTokenCount` be  `undefined`?

Reply:
Good point, I think these are gaps with my current implementation. The docs do define some [default values](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-text.html) for these optional parameters.

I'll update my implementation to default to these values and add the extra null check as well.

Reply:
Synced offline. Will just add the null check. No need to add default values.

Reply:
Related [changes](https://github.com/aws-observability/aws-otel-js-instrumentation/pull/111/files#r1825027958)